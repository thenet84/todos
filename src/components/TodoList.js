import React from 'react';
import PropTypes from 'prop-types';

import { TodoItem } from './TodoItem';

export const TodoList = ({ deleteTodo, markComplete, todos }) => (
  <div>
    {todos.map(todo => (
      <TodoItem
        key={todo.id}
        id={todo.id}
        title={todo.title}
        completed={todo.completed}
        deleteTodo={deleteTodo}
        markComplete={markComplete}
      />
    ))}
  </div>
);

TodoList.propTypes = {
  deleteTodo: PropTypes.func,
  markComplete: PropTypes.func,
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      completed: PropTypes.bool,
    })
  ),
};
