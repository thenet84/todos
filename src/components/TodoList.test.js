import React from 'react';
import { shallow } from 'enzyme';

import { TodoList } from './TodoList';

describe('TodoList Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TodoList todos={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
