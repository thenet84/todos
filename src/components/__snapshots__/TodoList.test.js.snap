// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component should render correctly 1`] = `<div />`;
